Silicon/CommonSocPkg/Library/WatchDogTimerLib/WatchDogTimerLib.c

```diff
@@ -238,3 +259,55 @@ IsWdtLocked (
   }
 }
 
+/**
+  Get the number of failed boots.
+
+  @retval UINT32              the number of boots
+
+**/
+UINT32
+EFIAPI
+GetFailedBootCount (
+  VOID
+  )
+{
+  return WdtGetScratchpad (B_ACPI_IO_OC_WDT_CTL_SCRATCHPAD_BOOT_CNT_MASK) >>
+                            B_ACPI_IO_OC_WDT_CTL_SCRATCHPAD_BOOT_CNT_SHIFT;
+}
+
+/**
+  Increment the number of failed boots.
+**/
+VOID
+EFIAPI
+IncrementFailedBootCount (
+  VOID
+  )
+{
+  UINT32 FailedBootCnt;
+
+  // Get current failed boot count
+  FailedBootCnt = GetFailedBootCount ();
+
+  // Clear current failed boot count
+  ClearFailedBootCount ();
```

Comment:
Does failed count bits need to be cleared before writing new value?

Reply:
If we want to leverage the existing WdtClearScratchpad() and WdtSetScratchpad() APIs, then yes. Please see https://github.com/slimbootloader/slimbootloader/pull/1647#discussion_r934769084 for a more detailed explanation.